src/notifications/settings/hooks.ts

```diff
@@ -32,21 +32,23 @@ export const useAllNotificationSettingsFromStorage = () => {
     useState<WalletNotificationSettings[]>(walletNotificationSettingsData);
   const [globalNotificationSettings, setGlobalNotificationSettings] = useState<GlobalNotificationTopics>(globalNotificationSettingsData);
   const [existingGroupSettings, setExistingGroupSettings] = useState<GroupSettings>(existingGroupSettingsData);
-  const listener = notificationSettingsStorage.addOnValueChangedListener(changedKey => {
-    if (changedKey === WALLET_TOPICS_STORAGE_KEY) {
-      const newSettings = notificationSettingsStorage.getString(changedKey);
-      newSettings && setWalletNotificationSettings(JSON.parse(newSettings));
-    } else if (changedKey === WALLET_GROUPS_STORAGE_KEY) {
-      const newSettings = notificationSettingsStorage.getString(changedKey);
-      newSettings && setExistingGroupSettings(JSON.parse(newSettings));
-    } else if (changedKey === GLOBAL_TOPICS_STORAGE_KEY) {
-      const newSettings = notificationSettingsStorage.getString(changedKey);
-      newSettings && setGlobalNotificationSettings(JSON.parse(newSettings));
-    }
-  });
-  useEffect(() => () => {
-    listener.remove();
-  });
+  useEffect(() => {
```

Comment:
needed to add this into the `useEffect` because of the async nature of listener changes, which was causing the returned `walletNotificationSettings` to not necessarily be in sync with the results after a `setWalletNotificationSettings`